### PR TITLE
Add build-id to binary output

### PR DESCRIPTION
**Pull request by jakubch1**
Without this change using binaries in nuget is failing:

/home/jakubch/.nuget/packages/microsoft.dotnet.arcade.sdk/6.0.0-beta.21125.5/tools/Publish.proj(236,5): error : Invalid ELF BuildID '<null>' for content/native/x64/libInstrumentationEngine.so

### CODE REVIEW

**Comment by WilliamXieMSFT** [comment]
Mind updating the ChangeLog for 1.0.38?

**Comment by jakubch1** [comment]
Updated CHANGELOG

**Review by WilliamXieMSFT — APPROVED** [review]
:shipit:

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -253,7 +253,7 @@ set(CMAKE_MACOSX_RPATH true)
 # Don't allow shared libraries to be built with undefined symbols
 # Mac builds complain about these, Linux needs the switch
 if(CMAKE_SYSTEM_NAME STREQUAL Linux)
-set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-allow-shlib-undefined" )
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-allow-shlib-undefined -Wl,--build-id=sha1" )
```
Comment (wiktork) [inline src/CMakeLists.txt:256]: is -Wl duplicated?

**Pull request merged**